src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,210 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import randoop.operation.TypedOperation;
+import randoop.util.Randomness;
+import randoop.util.SimpleArrayList;
+
+/**
+ * Implements the Bloodhound component, as described by the Guided Random Testing (GRT) paper.
+ * Bloodhound computes a weight for each method under test by taking a weighted combination of the
+ * uncovered branch ratio and the ratio between the number of times the method has been invoked and
+ * the maximum number of times any method under test has been invoked.
+ *
+ * <p>However, some hyper-parameters and edge cases were left unspecified in the GRT paper. We have
+ * chosen our own values for the following unspecified hyper-parameters:
+ *
+ * <ul>
+ *   <li>alpha - parameter to balance branch coverage and number of invocations when computing
+ *       weight.
+ *   <li>p - parameter for decreasing weights of methods between updates to coverage information.
+ *   <li>Interval for recomputing branch coverage information.
+ *   <li>Default weight for cases where a method has zero branches or computed weight is zero.
+ * </ul>
+ */
+public class Bloodhound implements TypedOperationSelector {
+
+  /**
+   * Map of methods under test to their weights. These weights are dynamic and depend on branch
+   * coverage.
+   */
+  private final Map<TypedOperation, Double> methodWeights = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have been recently selected by the {@link
+   * ForwardGenerator} to construct a new sequence. This map is cleared every time branch coverage
+   * is recomputed.
+   */
+  private final Map<TypedOperation, Integer> methodSelections = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have ever been selected by the {@link
+   * ForwardGenerator} to extend an existing sequence to construct a new sequence. This definition
+   * is the same as that of {@code methodSelections} except that we do not clear this map every time
+   * we recompute branch coverage. Thus, the integer value for a given method is non-decreasing
+   * during a run of Randoop. The GRT paper does not state its definition of the "number of
+   * invocations" of a method under test.
+   */
+  private final Map<TypedOperation, Integer> methodSuccessfulCalls = new HashMap<>();
```

Comment:
methodSelections and methodSuccessfulCalls represent the same thing.  Why is does one of them have the name "...Selections" and the other has the name "...Calls"?  Please make this consistent by changing one or both names.

Reply:
I had originally intended for the descriptions of the two variables in the GRT paper to be reflected in the variables' names.
![image](https://user-images.githubusercontent.com/15660492/38591412-7530a732-3ceb-11e8-93c3-cd9063767bc3.png)
However, I do agree that it would make more sense to rename the variables.
What do you think about:
`methodSelections` -> `currMethodSelectionCounts`
`methodSuccessfulCalls` -> `methodSelectionTotals`

Reply:
Thanks.  That text is very helpful.

I would guess that "selections" and "successful invocations" mean different things.  Here is my interpretation:
"Selections" means an attempt by Randoop to generate a test that culminates with the given method.
"Successful invocations" means that the call was classified as a normal execution (as opposed to being classified as fault-revealing or as invalid.

Do you think this is what the authors meant, or they were using the two distinct terms as synonyms?